specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json

```diff
@@ -0,0 +1,1956 @@
+          "format": "int64",
+          "description": "The current usage of the resource."
+        },
+        "limit": {
+          "readOnly": true,
+          "type": "integer",
+          "format": "int64",
+          "description": "The maximum permitted usage of the resource."
+        },
+        "name": {
+          "readOnly": true,
+          "$ref": "#/definitions/UsageName",
+          "description": "The name of the type of usage."
+        }
+      },
+      "description": "Describes AML Resource Usage."
+    },
+    "ListUsagesResult": {
+      "properties": {
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Usage"
+          },
+          "description": "The list of AML resource usages."
+        },
+        "nextLink": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The URI to fetch the next page of AML resource usage information. Call ListNext() with this to fetch the next page of AML resource usage information."
+        }
+      },
+      "description": "The List Usages operation response."
+    },
+    "VirtualMachineSize": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "title": "Virtual Machine size name",
+          "description": "The name of the virtual machine size."
+        },
+        "family": {
+          "type": "string",
+          "title": "Virtual Machine family name",
+          "description": "The family name of the virtual machine size."
+        },
+        "vCPUs": {
+          "type": "integer",
+          "format": "int32",
+          "title": "Number of vPUs",
+          "description": "The number of vCPUs supported by the virtual machine size."
+        },
+        "osVhdSizeMB": {
+          "type": "integer",
+          "format": "int32",
+          "title": "OS VHD Disk size",
+          "description": "The OS VHD disk size, in MB, allowed by the virtual machine size."
+        },
+        "maxResourceVolumeMB": {
+          "type": "integer",
+          "format": "int32",
+          "title": "Resource volume size",
+          "description": "The resource volume size, in MB, allowed by the virtual machine size."
+        },
+        "memoryGB": {
+          "type": "number",
+          "format": "double",
+          "title": "Memory size",
+          "description": "The amount of memory, in GB, supported by the virtual machine size."
+        },
+        "lowPriorityCapable": {
+          "type": "boolean",
+          "title": "Low priority capable",
+          "description": "Specifies if the virtual machine size supports low priority VMs."
+        },
+        "premiumIO": {
+          "type": "boolean",
+          "title": "Premium IO supported",
+          "description": "Specifies if the virtual machine size supports premium IO."
+        }
+      },
+      "description": "Describes the properties of a VM size."

```

Comment:
We can mark all these properties as readonly right?

Reply:
sure, we should be able to change it to readonly ...

Reply:
sure, we should be able to change it to readonly ...s